### PR TITLE
Preserve scopes in stream operations

### LINKED ISSUE #3975: Scopes are not being preserved by certain stream operators
Corresponding issue in ZIO with a test that fails with the current Effect version:

https://github.com/zio/zio/pull/8574

### PR DESCRIPTION
**Pull request by IMax153**
## Type

- [ ] Refactor
- [ ] Feature
- [x] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Related

- Related Issue #3975
- Closes #3975

### CODE REVIEW

File: packages/effect/src/internal/channel.ts [hunk 1]
```diff
@@ -1134,180 +1113,179 @@ export const mergeAllWith = (
...
-                      Effect.zipRight(raceEffects),
-                      withPermits(1),
-                      Effect.forkScoped
-                    )
-                    yield* $(Deferred.await(latch))
-                    const errored = yield* $(Deferred.isDone(errorSignal))
-                    return !errored
-                  })
+                      )
+                      // const raceEffects: Effect.Effect<void, OutErr | OutErr1, Env | Env1> = queueReader.pipe(
```
Comment (tim-smart) [inline packages/effect/src/internal/channel.ts:1216]: Can remove this?

**Review by tim-smart — COMMENTED** [review]
Seems OK. The diff is a bit large to scrutinize everything, but all the tests are passing so I'm ok merging this.

File: packages/effect/src/internal/channel.ts (hunk 1 above)
Comment (IMax153) [reply packages/effect/src/internal/channel.ts:1216]: My bad - yep will remove and merge.

**Comment by mikearnaldi** [comment]
LGTM!

**Pull request merged**